src/MongoDB/Manager.c

```diff
@@ -200,6 +200,15 @@ static void php_phongo_manager_prep_uri_options(zval *options TSRMLS_DC) /* {{{
 				continue;
 			}
 
+            if (!strcasecmp(ZSTR_VAL(string_key), MONGOC_URI_READPREFERENCE)) {
+				ZVAL_DEREF(option);
+				SEPARATE_ZVAL_NOREF(option);
+				if (Z_TYPE_P(option) != IS_STRING) {
+                    phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Unsupported readPreference type, a string must be given");
```

Comment:
after trying to move the responsibility of validating the `readpreference`option from php side to bson side , we will always got that exception thrown when we are trying to pass un-serializeable object like [ MongoDB\Driver\ReadPreference ]

> Serialization of 'MongoDB\Driver\ReadPreference' is not allowed

code to re-produce : 
```php
$manager = new MongoDB\Driver\Manager('mongodb://localhost:27017', [
    'readpreference' => new MongoDB\Driver\ReadPreference(MongoDB\Driver\ReadPreference::RP_SECONDARY),
]);
```

and I think that the problem is that we are proceeding the validation operation from checking an object as a `zval` to bson value

and that's mean that we will always be stuck in [here](https://github.com/mongodb/mongo-php-driver/blob/master/src/MongoDB/Manager.c#L270)
```c
if (options) {
    php_phongo_manager_prep_uri_options(options TSRMLS_CC);
}
```

and will not move forward to `phongo_manager_init(intern ....`